tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_Hierarchical.cpp

```diff
@@ -0,0 +1,2383 @@
+  result_tensor0[mesh.number_of_grid_points() - 1] = 1.0;
+  CHECK(get(get<::Tags::Modal<ScalarTag<0>>>(package_data0.volume_data)) ==
+        result_tensor0);
+  ModalVector result_tensor1(mesh.number_of_grid_points(), 0.0);
+  result_tensor1[mesh.number_of_grid_points() - 1] = 1.0;
+  result_tensor1[mesh.number_of_grid_points() - 2] = 2.0;
+  CHECK(get<0>(get<::Tags::Modal<VectorTag<1, 0>>>(
+            package_data0.volume_data)) == result_tensor1);
+}
+
+void two_neighbors() noexcept {
+  constexpr size_t dim = 1;
+  const size_t order = 3;
+  const Mesh<dim> mesh(order + 1, Spectral::Basis::Legendre,
+                       Spectral::Quadrature::GaussLobatto);
+  const size_t num_pts = mesh.number_of_grid_points();
+  const auto x = logical_coordinates(mesh);
+  using Limiter =
+      Hierarchical<dim, tmpl::list<ScalarTag<0>, VectorTag<dim, 0>>>;
+  // Use non-unity (because that's the default) but close alpha values to make
+  // the math easier but still test thoroughly.
+  Limiter hierarchical{
+      HierarchicalType::Krivodonova,
+      std::vector<double>(
+          Spectral::maximum_number_of_points<Spectral::Basis::Legendre>, 0.99)};
+
+  NeighborData<dim, typename Limiter::PackagedData> neighbor_data{};
+
+  const Element<dim> element(ElementId<dim>{0}, {});
+  // We don't care about the ElementId for these tests, just the direction.
+  Limiter::PackagedData& package_data_upper = neighbor_data[std::make_pair(
+      Direction<dim>::upper_xi(), ElementId<dim>{0})];
+  Limiter::PackagedData& package_data_lower = neighbor_data[std::make_pair(
+      Direction<dim>::lower_xi(), ElementId<dim>{0})];
+
+  package_data_upper.volume_data.initialize(num_pts);
+  package_data_upper.mesh = mesh;
+  package_data_lower.volume_data.initialize(num_pts);
+  package_data_lower.mesh = mesh;
+
+  Scalar<DataVector> self_lnc(num_pts, 0.0);
+  tnsr::I<DataVector, dim> self_vector(num_pts, 0.0);
+  DataVector expected(num_pts);
+  const auto helper =
+      [
+        &element, &expected, &hierarchical, &mesh, &neighbor_data,
+        &package_data_lower, &package_data_upper, &self_lnc, &self_vector, &x
+      ](const ModalVector& upper_coeffs,
+        const std::vector<double>& initial_coeffs,
+        const ModalVector& lower_coeffs,
+        const std::vector<double>& expected_coeffs) noexcept {
+    set_legendre(&get(self_lnc), mesh, x, initial_coeffs);
+    get(get<::Tags::Modal<ScalarTag<0>>>(package_data_upper.volume_data)) =
+        upper_coeffs;
+    get(get<::Tags::Modal<ScalarTag<0>>>(package_data_lower.volume_data)) =
+        lower_coeffs;
+    for (size_t i = 0; i < dim; ++i) {
+      set_legendre(&self_vector.get(i), mesh, x, initial_coeffs);
+      get<::Tags::Modal<VectorTag<dim, 0>>>(package_data_upper.volume_data)
+          .get(i) = upper_coeffs;
+      get<::Tags::Modal<VectorTag<dim, 0>>>(package_data_lower.volume_data)
+          .get(i) = lower_coeffs;
+    }
```

Comment:
I think it would be a much more interesting test to have a _single_ limiter call where the Scalar and Vector inputs have different values and therefore different logical paths. Here you make a large number of calls but because all the variable components are identical, the test can't find bugs mixing different tensor components (obviously unlikely).

I would suggest cleaning the 1D tests you have now to use just the Scalar, then add a single limiter invocation in which you test limiting both the Scalar and Vector with more non-identical inputs.

Reply:
I don't disagree that testing a call of the different limiters is helpful to test that the components are not mixed. The current test (which I'd like to keep) makes sure that both Scalar's and Tensor's have all their components limited. I think testing a single case for the limiter not mixing tensors/components and testing that in all cases the limiter is applied to all components is best.

I'll add the suggested test case :)